numba/core/typing/asnumbatype.py

```diff
@@ -31,15 +33,27 @@ def __init__(self):
 
         self.functions = [self._builtin_infer, self._numba_type_infer]
 
+        self.generic_superclasses = (py_typing._GenericAlias,)
+        self.union_superclasses = ()
+
+        # Support PEP 585 and PEP 604 on the relevant versions
+        if sys.version_info >= (3, 9):
```

Comment:
```suggestion
        if PYVERSION >= (3, 9):
```

And import `PYVERSION` from `numba.core.config`